index.js

```diff
@@ -1,81 +1,162 @@
-	return await fetch(`https://i.instagram.com/api/v1/media/configure_text_only_post/`, {
-		method: 'POST',
-		headers: {
-			'User-Agent': 'Barcelona 289.0.0.77.109 Android',
-			'Authorization': 'Bearer IGT:2:' + token,
-			'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-			'Sec-Fetch-Site': 'same-origin'
-		},
-		body: `signed_body=SIGNATURE.%7B%22publish_mode%22%3A%22text_post%22%2C%22text_post_app_info%22%3A%22%7B%5C%22reply_control%5C%22%3A0%7D%22%2C%22timezone_offset%22%3A%22-25200%22%2C%22source_type%22%3A%224%22%2C%22_uid%22%3A%22${userId}%22%2C%22device_id%22%3A%22android-${(Math.random()*1e24).toString(36)}%22%2C%22caption%22%3A%22${contents}%22%2C%22device%22%3A%7B%22manufacturer%22%3A%22OnePlus%22%2C%22model%22%3A%22ONEPLUS+A3010%22%2C%22android_version%22%3A25%2C%22android_release%22%3A%227.1.1%22%7D%7D`
-	}).then(async res => {
-		const user = await res.json()
-		return user
-	})
+class ThreadsAPI {
+  constructor({ token, userAgent, appId }) {
+    if (!token) {
+      throw new Error("Token is required.");
+    }
+
+    this.token = token;
+    this.userAgent = userAgent || "Barcelona 289.0.0.77.109 Android";
+    this.appId = appId || "238260118697367";
+  }
+
+  async getLsd() {
+    const url = "https://www.threads.net/@instagram";
+    const requestOptions = {
+      method: "GET",
+    };
+
+    const response = await fetch(url, requestOptions);
+    const text = await response.text();
+    const pos = text.search('"token"');
+    const end = text.substring(pos).search('"');
+    const lsd = text.substring(pos + 9, end - 1);
+    return lsd;
+  }
+
+  async getUser(userId) {
+    const url = `https://i.instagram.com/api/v1/users/${userId}/info`;
+    const requestOptions = {
+      headers: {
+        "User-Agent": this.userAgent,
+        Authorization: `Bearer IGT:2:${this.token}`,
+      },
+    };
+
+    const response = await fetch(url, requestOptions);
+    const user = await response.json();
+    return user;
+  }
+
+  async getPost(postId) {
+    const lsd = await this.getLsd();
+    const url = "https://www.threads.net/api/graphql";
+    const requestBody = {
+      lsd: lsd,
+      variables: JSON.stringify({ postID: postId }),
+      doc_id: "5587632691339264",
+    };
+
+    const requestOptions = {
+      method: "POST",
+      credentials: "omit",
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "X-IG-App-ID": this.appId,
+        "X-FB-LSD": lsd,
+        "Sec-Fetch-Site": "same-origin",
+      },
+      body: `lsd=${requestBody.lsd}&variables=${encodeURIComponent(
+        requestBody.variables
+      )}&doc_id=${requestBody.doc_id}`,
+    };
+
+    const response = await fetch(url, requestOptions);
+    return await response.json();
+  }
+
+  async postThread(contents, userId) {
+    const id = (Math.random() * 1e24).toString(36);
+    const url =
+      "https://i.instagram.com/api/v1/media/configure_text_only_post/";
+
+    const requestBody = {
```

Comment:
afaik, decoding the request body causes weird issues in fetch. have you tested this function yet?